components/terrain/quadtreeworld.cpp

```diff
@@ -496,39 +500,39 @@ View* QuadTreeWorld::createView()
 void QuadTreeWorld::preload(View *view, const osg::Vec3f &viewPoint, const osg::Vec4i &grid, std::atomic<bool> &abort, Loading::Reporter& reporter)
 {
     ensureQuadTreeBuilt();
+    const float cellWorldSize = mStorage->getCellWorldSize();
 
     ViewData* vd = static_cast<ViewData*>(view);
     vd->setViewPoint(viewPoint);
     vd->setActiveGrid(grid);
-    DefaultLodCallback lodCallback(mLodFactor, mMinSize, mViewDistance, grid);
-    mRootNode->traverseNodes(vd, viewPoint, &lodCallback);
-
-    std::size_t progressTotal = 0;
-    for (unsigned int i = 0, n = vd->getNumEntries(); i < n; ++i)
-        progressTotal += vd->getEntry(i).mNode->getSize();
-
-    reporter.addTotal(progressTotal);
 
-    const float cellWorldSize = mStorage->getCellWorldSize();
-    for (unsigned int i=0; i<vd->getNumEntries() && !abort; ++i)
+    for (unsigned int pass=0; pass<3; ++pass)
     {
-        ViewData::Entry& entry = vd->getEntry(i);
-
-
+        unsigned int startEntry = vd->getNumEntries();
 
-        loadRenderingNode(entry, vd, mVertexLodMod, cellWorldSize, grid, mChunkManagers, true, mViewDataMap->getReuseDistance());
-        reporter.addProgress(entry.mNode->getSize());
+        float distanceModifier=0.f;
+        if (pass>0) distanceModifier = (pass==1) ? 1024 : -1024;
```

Comment:
This could be replaced with something like this, to improve clarity:

```cpp
if (pass == 1)
  distanceModifier = 1024;
else if (pass == 2)
  distanceModifier = -1024;
```